seasmon_xr/accessors.py

```diff
@@ -557,6 +557,12 @@ def mean(
         if not isinstance(zones, xarray.DataArray):
             raise ValueError("Zones need to be xarray.DataArray!")
 
+        if "nodata" not in zones.attrs:
+            raise ValueError("Zones xarray DataArray needs nodata attribute")
+
+        # set null values to nodata value
+        xx = xx.where(~xx.isnull(), xx.nodata).astype(xx.dtype)
```

Comment:
@Kirill888 would this suffice to account for `nan` in float input? The `nodata` attribute is required above.

Wasn't sure if we need to call `astype`, but I assumed this would keep int data in int after `where`

Reply:
@Kirill888 even though the PR is approved, please check above

Reply:
This looks fine, but I think xarray keeps original `dtype`, so not sure if `.astype` is needed here

Reply:
If it does not then maybe need to wrap `xx.dtype.type(xx.nodata)`

Reply:
actually `xx.notnull()` is probably more performant than `~xx.isnull()`